backhand-cli/src/bin/add.rs

```diff
@@ -103,8 +103,8 @@ fn main() -> ExitCode {
     }
 
     // write new file
-    let output = File::create(&args.out).unwrap();
-    if let Err(e) = filesystem.write(output) {
+    let mut output = File::create(&args.out).unwrap();
```

Comment:
@rbran I know you removed this requirement with https://github.com/wcampbell0x2a/backhand/pull/361.

 But the new proposed deku API writer (by myself) needs `&'a mut W`: https://github.com/sharksforarms/deku/pull/355/files#diff-7d5c45da93a71cc9d454396255360338e2476d783dcd17e9b57a09c4b36aa182R28. Or at least, I assume this is the best API.

I think you are more familiar with lifetimes than I am, do you have any comments or workarounds for this change?

Reply:
In both cases you don't need to restrict the parameter to only "mut ref" with `w: &mut W`, you can allow both "ref" and "owned" with `mut w: W`. When passing as parameter you can use `a.fun(w)` to move the value or `a.fun(&mut w)` to borrow the value.

Reply:
I did a PR to your PR LOL: https://github.com/wcampbell0x2a/backhand/pull/386

Reply:
I did the same with deku, but I'm not so sure about it, because I didn't take the proper time to get used with the macros: https://github.com/sharksforarms/deku/pull/384

Reply:
Thanks for the explanation and patches!